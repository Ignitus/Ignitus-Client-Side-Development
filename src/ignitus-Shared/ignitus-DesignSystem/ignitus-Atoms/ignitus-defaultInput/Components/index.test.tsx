import React from 'react';
import { mount, render } from 'enzyme';

import { DefaultInput } from './index';

const props = {
  placeholder: 'Placeholder',
  type: 'text',
  name: 'Input',
  handleChange: jest.fn(),
};

const input = type => `input[type="${type}"]`;

it('should rendered', () => {
  const tree = render(<DefaultInput {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultInput />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DefaultInput {...props} />);
  });

  it('should render input with defined type in prop', () => {
    expect(wrapper.find(input('text')).length).toBe(1);

    wrapper = mount(<DefaultInput {...{ ...props, type: 'password' }} />);
    expect(wrapper.find(input('password')).length).toBe(1);
  });

  it('should handle input change event properly', () => {
    wrapper
      .find(input(props.type))
      .simulate('change', { target: { value: 'value' } });

    expect(props.handleChange).toBeCalledWith('value');
  });
});
