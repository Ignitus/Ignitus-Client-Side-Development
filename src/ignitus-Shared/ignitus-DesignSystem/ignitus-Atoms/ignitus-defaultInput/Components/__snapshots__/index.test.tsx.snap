// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
<input
  class="css-8xbheg"
  name="Input"
  placeholder="Placeholder"
  type="text"
/>
`;
